### PR TITLE
added fix for virtual order (giftcards only orders)

### PR DESCRIPTION
**Pull request by zainengineer**
added fix for virtual order (giftcards only orders)
without the fix a fatal error will occur

### CODE REVIEW

**Comment by zainengineer** [comment]
@mattwellss please merge this

File: app/code/community/QuBit/UniversalVariable/Model/Page/Observer.php [hunk 1]
```diff
@@ -529,11 +529,13 @@ public function _setTranscation() {
       $transaction['shipping_cost']   = (float) $order->getShippingAmount();
       $transaction['shipping_method'] = $this->_extractShippingMethod($order);
 
-      // Get addresses
-      if (method_exists($order,'getShippingAddress')) {
+      //checking for getShippingAddress() otherwise $order->getShippingAddress()->getId() will result in fatal error
+      //if order is all virtual i.e. only gift cards are purchased.
+      if ((method_exists($order,'getShippingAddress') && $order->getShippingAddress())) {
```
Comment (mattwellss) [inline app/code/community/QuBit/UniversalVariable/Model/Page/Observer.php]: @zainengineer Is the `method_exists` call is necessary?

**Comment by zainengineer** [comment]
@mattwellss  `method_exists` check was already there in the code

**Comment by mattwellss** [comment]
:+1:

**Comment by mattwellss** [comment]
@parhamr if you have the chance, this can be merged!

**Comment by mattwellss** [comment]
@parhamr ping!!

**Pull request merged**